tests/golang.rs

```diff
@@ -24,7 +25,92 @@ macro_rules! tests {
 
         // https://github.com/golang/go/blob/master/test/chan/doubleselect.go
         mod doubleselect {
-            // TODO
+            use super::*;
+
+            const ITERATIONS: usize = 100_000;
+
+            #[test]
+            #[ignore]
+            fn test_double_select() {
+                let (c1, r1) = channel::unbounded();
+                let (c2, r2) = channel::unbounded();
+                let (c3, r3) = channel::unbounded();
+                let (c4, r4) = channel::unbounded();
+                let (done_s, done_r) = channel::unbounded();
+                let (mux_s, mux_r) = channel::unbounded();
+
+                fn mux(out: channel::Sender<usize>, in_c: channel::Receiver<usize>, done: channel::Sender<bool>) {
+                    for val in in_c.0 {
+                        out.send(val);
+                    }
+
+                    done.send(true);
+                }
+
+                crossbeam::scope(|scope| {
+                    // This is akin to the sender function in the go example.
+                    scope.spawn(|| {
+                        for i in 0..ITERATIONS {
+                            select! {
+                                send(c1, i) => {},
+                                send(c2, i) => {},
+                                send(c3, i) => {},
+                                send(c4, i) => {},
+                            }
+                        }
+
+                        drop(c1);
+                        drop(c2);
+                        drop(c3);
+                        drop(c4);
+                    });
+
+                    // This is akin to the mux function call.
+                    scope.spawn(|| {
+                        let mux_s = mux_s.clone();
+                        let done_s = done_s.clone();
+                        mux(mux_s, r1, done_s);
+                    });
+                    scope.spawn(|| {
+                        let mux_s = mux_s.clone();
+                        let done_s = done_s.clone();
+                        mux(mux_s, r2, done_s);
+                    });
+                    scope.spawn(|| {
+                        let mux_s = mux_s.clone();
+                        let done_s = done_s.clone();
+                        mux(mux_s, r3, done_s);
+                    });
+                    scope.spawn(|| {
+                        let mux_s = mux_s.clone();
+                        let done_s = done_s.clone();
+                        mux(mux_s, r4, done_s);
+                    });
+
+                    scope.spawn(|| {
+                        let mux_s = mux_s.clone();
```

Comment:
We should just move `mux_s` into this closure and drop it at the end rather than capturing it by reference and making a clone of it.

Reply:
Won't this throw a borrow check error? Since the other threads will need to reference `mux_s`? Unless I am misunderstanding you.

Reply:
Ah, thanks for sharing. I never remember to scope values like that. I updated it and removed the ignore annotation.